### PR TITLE
Add Interaction#getUserLocale and Interaction#getGuildLocale

### PR DESCRIPTION
**Pull request by caneleex**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [x] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

this PR adds `Interaction#getUserLocale` and `Interaction#getGuildLocale`. see https://github.com/discord/discord-api-docs/pull/4265. I'm not too sure about the behavior of `getGuildLocale` because there's a separate field for the preferred guild locale (however that is still only changeable with the `COMMUNITY` feature, effectively being `Guild#getLocale`), so feel free to comment on that.

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/internal/interactions/InteractionImpl.java [hunk 1]
```diff
@@ -83,6 +88,8 @@ public InteractionImpl(long id, int type, String token, Guild guild, Member memb
         this.member = member;
         this.user = user;
         this.channel = channel;
+        this.userLocale = userLocale;
+        this.guildLocale = guildLocale;
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/internal/interactions/InteractionImpl.java:92]: `guildLocale` is unneeded

File: src/main/java/net/dv8tion/jda/internal/interactions/InteractionImpl.java [hunk 2]
```diff
@@ -120,6 +127,20 @@ public Channel getChannel()
...
+    {
+        return userLocale;
+    }
+
+    @Nonnull
+    @Override
+    public Locale getGuildLocale()
+    {
+        return guildLocale;
+    }
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/internal/interactions/InteractionImpl.java:142]: This is implemented as a `default` method in the interface and is unneeded

**Review by DV8FromTheWorld — CHANGES_REQUESTED** [review]

File: src/main/java/net/dv8tion/jda/api/interactions/Interaction.java [hunk 3]
```diff
@@ -503,6 +504,30 @@ default PrivateChannel getPrivateChannel()
         throw new IllegalStateException("Cannot convert channel of type " + getChannelType() + " to PrivateChannel");
     }
 
+    /**
+     * Returns the selected language of the invoking user.
+     * This is null if the {@link InteractionType type} of this interaction is {@link InteractionType#PING PING}.
+     *
+     * @return The language of the invoking user or null if the type is {@link InteractionType#PING PING}
```
[the remark this answers is not in the document]
Comment (caneleex) [reply src/main/java/net/dv8tion/jda/api/interactions/Interaction.java:511]: should I just make this Nonnull until receiving PING is added?

[the remark this answers is not in the document]
Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/api/interactions/Interaction.java:511]: PING is only for webhook interactions which we don't, currently don't plan to, support

**Comment by MinnDevelopment** [comment]
Could you rebase and target this onto the `rework/interactions` branch instead?

**Comment by caneleex** [comment]
I fucked up the rebase. I'll open a new pr.

**Pull request closed without merging**